server/heroku/heroku.go

```diff
@@ -169,6 +170,15 @@ func UserFromContext(ctx context.Context) *empire.User {
 	return u
 }
 
-func findMessage(r *http.Request) (string, error) {
-	return r.Header.Get(heroku.CommitMessageHeader), nil
+type shouldRequireMessages interface {
+	ShouldRequireMessages() bool
+}
+
+func findMessage(r *http.Request, e shouldRequireMessages) (string, error) {
+	var err error
+	h := r.Header.Get(heroku.CommitMessageHeader)
+	if e.ShouldRequireMessages() && h == "" {
+		err = fmt.Errorf("Header '%s' is required", heroku.CommitMessageHeader)
```

Comment:
I was curious if this was the best way to return an error. I'll handle this within emp and with the current error just search for the string "CommitMessage" and required or something, but wasn't sure if there was a better way to do this.